sw/device/lib/dif/dif_kmac_unittest.cc

```diff
@@ -0,0 +1,108 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_kmac.h"
+
+#include <string>
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace dif_kmac_unittest {
+namespace {
+using ::testing::ElementsAre;
+
+/**
+ * Decode a string encoded as:
+ *
+ *   left_encode(len(s)) || s
+ *
+ * For more details see 'NIST SP 800-185: 2.3.2 String Encoding'.
+ */
+std::string DecodeBitString(const char *buffer) {
+  // Number of bytes used to encode the length.
+  uint8_t n = static_cast<uint8_t>(*buffer++);
+
+  // Decode the length (big-endian integer `n` bytes long).
+  uint64_t len = 0;
+  EXPECT_TRUE(n <= sizeof(len));
+  for (uint8_t i = 1; i <= n; ++i) {
+    uint8_t xi = static_cast<uint8_t>(*buffer++);
+    len |= static_cast<uint64_t>(xi) << ((n - i) * 8);
+  }
+
+  // Convert len from bits to bytes.
+  EXPECT_TRUE(len % 8 == 0);
+  len /= 8;
+
+  return std::string(buffer, static_cast<size_t>(len));
+}
+
+TEST(CustomizationStringTest, Encode) {
+  dif_kmac_customization_string_t cs;
+
+  EXPECT_EQ(dif_kmac_customization_string_init(nullptr, 0, &cs), kDifKmacOk);
+  EXPECT_EQ(DecodeBitString(cs.buffer), "");
```

Comment:
I know what you're going for ehre, but do you think it makes sense to do a *manual* comparision, including the leading tag and BE length?

Reply:
Yeah, I was in two minds about this one. I've changed it to something more manual.